tests/integration/pilot/gateway_conformance_test.go

```diff
@@ -79,6 +86,48 @@ func TestGatewayConformance(t *testing.T) {
 		Run(func(ctx framework.TestContext) {
 			DeployGatewayAPICRD(ctx)
 
+			// Precreate the GatewayConformance namespaces, and apply the Image Pull Secret to them.
+			if ctx.Settings().Image.PullSecret != "" {
+				for _, ns := range conformanceNamespaces {
+					namespace.Claim(ctx, namespace.Config{
+						Prefix: ns,
+						Inject: false,
+					})
+					// Cannot apply the existing secret directly; need to sanitize the
```

Comment:
Why not? We do it in `(n *kubeNamespace) createInCluster`

Reply:
This seems like the input is just wrong...?

Reply:
The specific error I encountered was something like:
```
operation cannot be fulfilled on secrets: the object has been modified; please apply your changes to the latest version and try again
```

From what I understand, this happens when you apply a resource with a resourceVersion in its metadata that differs from the original. In this case, the resource that the PullSecret points to appears to already have that data in it, so to bypass the issue I import it as an object, sanitize everything but the name from the metadata, and then reapply it in the right namespace.

Reply:
Why don't we just point to a PullSecret that is valid already?

Reply:
Actually, it may be that createInCluster creates the same secret already, so the apply I was making was failing because it had a different resourceVersion specified. Let me check if the rewrite fixes it.

Reply:
Fixed in rewrite.